_posts/2020-07-12-ray-tune.md

```diff
@@ -0,0 +1,234 @@
+---
+layout: post
+title: Dask-ML offers modern hyperparameter optimization techniques packaged in a Scikit-learn API
+tagline: This has been implemented in Dask-ML for over a year
+author: <a href="https://stsievert.com">Scott Sievert</a> (University of Wisconsin–Madison)
+tags: [machine-learning, dask-ml, dask, ray]
+theme: twitter
+---
+
+{% include JB/setup %}
+
+Hyperparameter optimization is the process of deducing machine learning model
+parameters that can't learned from data. These "hyperparameters" have to be
+given at model initialization. An example is with the `alpha` parameter in
+Scikit-learn's [Ridge]. For a more complete overview, see "[How to Use t-SNE
+Effectively][tsne]" or "[Tuning the hyper-parameters of an estimator][tuning]."
+
+[The introduction][1] of Ray's Scikit-learn compatible hyperparameter
+optimization library made an counterfactual statement:
+
+[tsne]:https://distill.pub/2016/misread-tsne/
+[tuning]:https://scikit-learn.org/stable/modules/grid_search.html
+
+[Ridge]:https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.Ridge.html#sklearn.linear_model.Ridge
+[1]:https://medium.com/distributed-computing-with-ray/gridsearchcv-2-0-new-and-improved-ee56644cbabf
+
+> Cutting edge hyperparameter tuning techniques (bayesian optimization, *early
+> stopping, distributed execution*) can provide significant speedups over grid
+> search and random search.
+>
+> However, the machine learning ecosystem is missing a solution that provides
+> users with the ability to leverage these new algorithms while allowing users
+> to stay within the Scikit-Learn API. In this blog post, we introduce
+> tune-sklearn to bridge this gap. Tune-sklearn is a drop-in replacement for
+> Scikit-Learn’s model selection module with state-of-the-art optimization
+> features. [emphasis added]
+>
+> —[GridSearchCV 2.0 — New and Improved][1]
+
+This is a blatantly misleading: Dask-ML fills the same gap they claim to bridge.
```

Comment:
Agreed that we'll want to moderate the tone here and elsewhere :)

Reply:
:+1: